lab-collin/src/main.js

```diff
@@ -0,0 +1,124 @@
+import React from 'react';
+import { render as reactDomRender } from 'react-dom'; 
+import superagent from 'superagent';
+import './style/main.scss';
+
+
+class SearchForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      searchFormBoard: '',
+      searchFormLimit: 0,
+    };
+
+    this.handleBoardChange = this.handleBoardChange.bind(this);
+    this.handleLimitChange = this.handleLimitChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleBoardChange(event) {
+    this.setState({ searchFormBoard: event.target.value });
+  }
+
+  handleLimitChange(event) {
+    this.setState({ searchFormLimit: event.target.value });
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    this.props.boardSelect(this.state.searchFormBoard, this.state.searchFormLimit);
+  }
+
+  render() {
+    return (
+      <form onSubmit={this.handleSubmit}>
+        <input 
+          type="text"
+          name="searchFormBoard"
+          placeholder="look up a reddit board"
+          value={this.state.searchFormBoard}
+          onChange={this.handleBoardChange}
+        />
+        <input 
+          type="number"
+          name="searchFormLimit"
+          placeholder="set result limit"
+          value={this.state.searchFormLimit}
+          onChange={this.handleLimitChange}
+        />
+         <input type="submit" value="Submit" />
+      </form>
+    );
+  }
+}
+
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      topics: [],
+    };
+
+    this.boardSelect = this.boardSelect.bind(this);
+  }
+
+  boardSelect(board, limit) {
+    return superagent.get(`http://reddit.com/r/${board}.json?limit=${limit}`)
```

Comment:
needs to be `http:// www.reddit.com....`, getting CORS error without

Also, you could set up a ternary statement just before the superagent request to check if the limit value is between 0 and 100, if not, render error message